uxarray/grid.py

```diff
@@ -225,3 +255,84 @@ def __init_ds_var_names__(self):
             "nMesh2_face": "nMesh2_face",
             "nMaxMesh2_face_nodes": "nMaxMesh2_face_nodes"
         }
+
+    def integrate(self, var_key):
+        """ Integrates over all the faces of the given mesh.
+        Parameters
+        ----------
+
+        var_key : string, required
+            Name of variable for integration.
+
+        Returns
+        -------
+
+        double: integration result.
+
+        Examples
+        --------
+
+        Open grid file only
+        >>> grid = ux.open_dataset("grid.ug", "centroid_pressure_data_ug")
+
+
+        Open grid file along with data
+        >>> integral_psi = grid.integrate("psi")
+        """
+        integral = 0.0
+
+        # call function to get area of all the faces as a np array
+        face_areas = self.face_areas
+
+        face_vals = self.ds.get(var_key).to_numpy()
+        integral = np.dot(face_areas, face_vals)
+
+        return integral
+
+    @property
```

Comment:
Regarding this usage, I have created a suggestion PR from a branch off this branch.